### PR TITLE
MCP.1 w/ refactor

### PR DESCRIPTION
**Pull request by TaranFurnell**
Refactor of MCP.1
Based on current FW MCP.1

There are a couple todos which will be addressed soon, but please start review now. All Google changes have been included in a single commit to simplify review.

### CODE REVIEW

**Comment by kate-harrison** [comment]
FYI this fixes the problems reported in Issue #595.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 1]
```diff
@@ -145,45 +159,65 @@ def generate_MCP_1_default_config(self, filename):
...
         'registrationRequests': [device_3],
-        'grantRequests': [grant_request_3]
+        'grantRequests': [grant_request_3],
+        'conditionalRegistrationData': []
     }
     cbsd_records_iteration_1_domain_proxy_0 = {
         'registrationRequests': [device_4],
-        'grantRequests': [grant_request_4]
+        'grantRequests': [grant_request_4],
+        'conditionalRegistrationData': [conditionals_device_4]
```
Comment (jni2000) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: why this different from line 197 and 207 ?

Comment (kate-harrison) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: What difference do you see?

**Comment by jni2000** [comment]
will this PR replace PR #588 ?

**Comment by kate-harrison** [comment]
Yes, this will replace #588 completely.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py (hunk 1 above)
Comment (kate-harrison) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: This CBSD had conditional registration data and the others didn't.

Comment (jni2000) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: ok, got it. Thanks.

**Comment by jni2000** [comment]
We are reviewing the changes, focusing on MCP.  Others look good to me so far. if someone else can help double check, it will be more secure.

**Comment by jni2000** [comment]
Finished a second round of review. Most of the changes are added for conditional data, introduced new class, logic change for IAP and Aggregate outputs store and compare. Only one comment, please take a look. If the code has been tested and run, they are good to go.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 2]
```diff
@@ -129,7 +141,9 @@ def generate_MCP_1_default_config(self, filename):
     esc_record_1 = json.load(
       open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
 
-    # Load PPA record
+    esc_record_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'esc_sensor_record_1.json')))
```
Comment (jni2000) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: Record “esc_sensor_record_1.json” is not present

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 3]
```diff
@@ -12,20 +12,33 @@
...
+from reference_models.iap import iap
+from reference_models.interference import aggregate_interference, interference
 
 
-class MultiConstraintProtectionTestcase(sas_testcase.SasTestCase):
+DELTA_IAP = 1 # Threshold value in dBm
+ONE_MHZ = 1000000
+
+
+class MultiConstraintProtectionTestcase(McpXprCommonTestcase):
```
Comment (jni2000) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: run time error :  “ undefined class”

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 4]
```diff
@@ -309,239 +347,447 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
...
-      for gwpz_record in protected_entity_records['gwpzRecords']:
+    if 'gwpzRecords' in self.protected_entity_records:
+      for gwpz_record in self.protected_entity_records['gwpzRecords']:
         self._sas_admin.InjectWisp({'record': gwpz_record})
+        # TODO: calculate and store the land category of the GWPZ
 
-    if 'escRecords' in protected_entity_records:
-      for esc_record in protected_entity_records['escRecords']:
+    if 'escRecords' in self.protected_entity_records:
+      for esc_record in self.protected_entity_records['escRecords']:
```
Comment (jni2000) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: run time error : “TypeError: list indices must be integers, not str”

File: src/harness/testcases/WINNF_FT_S_FPR_testcase.py [hunk 5]
```diff
@@ -213,7 +214,7 @@ def generate_FPR_1_default_config(self, filename):
     dump_records_iteration_0_sas_test_harness_1 = {
         'cbsdRecords': cbsd_fad_records_iteration_0_sas_test_harness_1
     }
- 
+
 
     # Create the actual config.
     iteration0_config = {
```
Comment (jni2000) [inline src/harness/testcases/WINNF_FT_S_FPR_testcase.py]: line: 225  Please   update  to “'clientCert': sas.GetDefaultDomainProxySSLCertPath()”
line: 226  Please   update  to “ 'clientKey':   sas.GetDefaultDomainProxySSLKeyPath() “

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 6]
```diff
@@ -309,239 +347,447 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
...
+    self.performAggregateInterferenceCheck()
+
+  def performIap(self):
+    self.ppa_ap_iap_ref_values_list = []
+    self.gwpz_ap_iap_ref_values_list = []
+    self.fss_blocking_ap_iap_ref_values_list = []
+    self.fss_cochannel_ap_iap_ref_values_list = []
+    self.esc_ap_iap_ref_values_list = []
+
+    authorized_grants = data.getAuthorizedGrantsFromDomainProxies(self.domain_proxy_objects)
```
[the remark this answers is not in the document]
Comment (TaranFurnell) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: Removed.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py (hunk 3 above)
Comment (TaranFurnell) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: Code has been reordered, this makes it a somewhat painful diff to read sorry

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py (hunk 4 above)
Comment (TaranFurnell) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: This was most likely an issue with how the protected_entity_records were being merged

File: src/harness/testcases/WINNF_FT_S_FPR_testcase.py (hunk 5 above)
Comment (TaranFurnell) [reply src/harness/testcases/WINNF_FT_S_FPR_testcase.py]: This was resolved with the merge from MCP.1-fix.

**Comment by jni2000** [comment]
Thanks, I think we are good to move forward now. Thanks for your great effort.

**Pull request merged**